src/system/displays/displays_windows.cpp

```diff
@@ -14,39 +14,42 @@
 #include <windows.h>
 
 
-std::vector<iware::system::display_t> iware::system::displays() {
-	const struct bundle {
-		HDC desktop_dc;
-		std::vector<iware::system::display_t> ret;
-	} bundle{GetDC(nullptr), {}};
-	iware::detail::quickscope_wrapper desktop_dc_deleter{[&]() { ReleaseDC(nullptr, bundle.desktop_dc); }};
+struct bundle {
```

Comment:
as i understand the error, is this just an attribute(stdcall) mismatch? would passing `(MONITORENUMPROC)([](auto, ...)` as an argument instead work?

Reply:
No, although I do not know the solid reason for it and I did not dig into it. Both c-style and reinterpret cast fail with "invalid cast":
```
/home/aki/src/infoware/src/system/displays/displays_windows.cpp: In function ‘std::vector<iware::system::display_t> iware::system::displays()’:
/home/aki/src/infoware/src/system/displays/displays_windows.cpp:26:13: error: invalid cast from type ‘iware::system::displays()::<lambda(auto:1, auto:2, auto:3, auto:4)>’ to type ‘MONITORENUMPROC’ {aka ‘int (__attribute__((stdcall)) *)(HMONITOR__*, HDC__*, tagRECT*, long int)’}
   26 |             reinterpret_cast<MONITORENUMPROC>([](auto, auto hdc, auto rect, auto userdata) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   27 |                     auto& bundle = *reinterpret_cast<struct bundle*>(userdata);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   28 |
      |
   29 |                     const unsigned int desktop_dpi = GetDeviceCaps(bundle.desktop_dc, LOGPIXELSX);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   30 |                     // https://blogs.msdn.microsoft.com/oldnewthing/20101013-00/?p=12543
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                     const unsigned int desktop_bpp    = GetDeviceCaps(bundle.desktop_dc, BITSPIXEL) * GetDeviceCaps(bundle.desktop_dc, PLANES);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                     const double desktop_refresh_rate = GetDeviceCaps(bundle.desktop_dc, VREFRESH);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   33 |
      |
   34 |
      |
   35 |                     // Sometimes returns 0 – fall back to the desktop's globals if so.
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   36 |                     const unsigned int monitor_dpi    = GetDeviceCaps(hdc, LOGPIXELSX);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   37 |                     const unsigned int monitor_bpp    = GetDeviceCaps(hdc, BITSPIXEL) * GetDeviceCaps(hdc, PLANES);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   38 |                     const double monitor_refresh_rate = GetDeviceCaps(hdc, VREFRESH);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   39 |
      |
   40 |                     const unsigned int width  = std::abs(rect->right - rect->left);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   41 |                     const unsigned int height = std::abs(rect->bottom - rect->top);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   42 |
      |
   43 |                     // See http://stackoverflow.com/a/12654433/2851815 and up for DPI. In short: can't be done too too well, go with best solution.
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   44 |                     bundle.ret.push_back({width, height, monitor_dpi ? monitor_dpi : desktop_dpi, monitor_bpp ? monitor_bpp : desktop_bpp,
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   45 |                                           monitor_refresh_rate ? monitor_refresh_rate : desktop_refresh_rate});
      |                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   46 |
      |
   47 |                     return TRUE;
      |                     ~~~~~~~~~~~~
   48 |             }),
      |             ~~
make[2]: *** [CMakeFiles/infoware.dir/build.make:617: CMakeFiles/infoware.dir/src/system/displays/displays_windows.cpp.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:94: CMakeFiles/infoware.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

With:
```
$ i686-w64-mingw32-g++ --version
i686-w64-mingw32-g++ (GCC) 13.1.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```